test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala

```diff
@@ -81,6 +81,64 @@ object OptimizerTest extends JasmineTest {
 
   }
 
+  describe("String_+ constant folding") {
+    it("must fold two constant string to the correct value") {
+      expect("I am " + "constant").toEqual("I am constant")
+    }
+
+    it("must fold the empty string when associated with a string") {
+      val number: String = "1" + (1.4f).toString.substring(1)
```

Comment:
It seems that this line is unnecessarily complex wrt. the title of that test. Why not simply `"hello"` or any other arbitrary string? To prevent it to be known by the optimizer, you can extract it in a `@noinline def str = "hello"`.